src/execution.ts

```diff
@@ -232,7 +240,17 @@ export function compileQuery(
       context.operation.variableDefinitions || []
     );
 
-    const functionBody = compileOperation(context);
+    const type = getOperationRootType(context.schema, context.operation);
+    const fieldMap = collectFields(
+      context,
+      type,
+      context.operation.selectionSet,
+      Object.create(null),
+      Object.create(null)
+    );
+
+    const functionBody = compileOperation(context, type, fieldMap);
+
     const compiledQuery: InternalCompiledQuery = {
```

Comment:
This is not supposed to be there in case of subscriptions, right?

Reply:
`compiledQuery.query()` is still be there. It will be used in `mapSourceToResponse` function similar to the way `graphql-js` uses `execute`. (with `rootValue` being the source)

https://github.com/graphql/graphql-js/blob/main/src/subscription/subscribe.js#L143